nats.go

```diff
@@ -2377,8 +2390,7 @@ func (nc *Conn) sendConnect() error {
 				// in doReconnect()).
 				nc.processAuthError(authErr)
 			}
-
-			return errors.New("nats: " + proto)
+			return &natsProtoErr{proto}
```

Comment:
Sure, but have you tried adding a `return authErr` in the "if" statement above? Does that work? Point is that we may not need to do any conversions for the known errors handled in checkAuthError. Your changes is still valid I guess for all other cases?

Reply:
And by "does it work", I mean the user code could then do `if err == nats.ErrAuthorization` without even have to use the errors.Is().

Reply:
It does work but then I saw that we also tested for uppercase literal error in other tests like: https://github.com/nats-io/nats.go/blob/main/test/cluster_test.go#L192-L194
and recommendation from Go team is to start using `errors.Is` anyway instead of checking directly to check for wrapped errors as well.